docs/python/generators/examples_helpers.py

```
@@ -146,10 +146,22 @@ def dedupe_examples_list(exs: List[Dict[str, Any]]) -> List[Dict[str, Any]]:
-    """Return mkdocs hl_lines string from snippet and tokens, or None."""
+    """Return mkdocs hl_lines string from snippet and tokens, or None.
+    
+    When a matched token line contains '{', extends highlighting to cover
+    the full brace-delimited block (up to the matching '}').
+    """
     if not snippet or not tokens:
         return None
     lines = snippet.splitlines()
+    # Strip leading blank lines so hl_lines matches rendered HTML numbering
+    first_nonblank = 0
+    for l in lines:
```

Ruff (0.15.21)

[error] Ambiguous variable name: `l`

(E741)